### PR TITLE
Extracting tasks and commandline arguments

### PR DESCRIPTION
**Pull request by cdsap**
Extracting components like tasks and commandLine providers from RoomWorkaround

### CODE REVIEW

**Review by runningcode — APPROVED** [review]
Looks good. I think I already reviewed this and there are no logic changes, just moving things out, right?

**Comment by cdsap** [comment]
yep, just extracting code from the RoomWorkaround class

File: src/main/groovy/org/gradle/android/workarounds/room/RoomExtension.groovy [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.gradle.android.workarounds.room.task.MergeAssociations
+import org.gradle.api.file.Directory
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+
+import javax.inject.Inject
+
+abstract class RoomExtension {
```
Comment (runningcode) [inline src/main/groovy/org/gradle/android/workarounds/room/RoomExtension.groovy:11]: let's add the `@CompileStatic` flag on this class and all the other added classes here if possible!

Comment (cdsap) [reply src/main/groovy/org/gradle/android/workarounds/room/RoomExtension.groovy:13]: added `@CompileStatic` in all the new classes

File: src/main/groovy/org/gradle/android/workarounds/room/task/RoomSchemaLocationMergeTask.groovy [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * during this execution, even if they are incomplete (they really shouldn't be, though).
+ *
+ * We don't want to create task dependencies on the compile/kapt tasks because we don't want to force execution
+ * of those tasks if only a single variant is being assembled.
+ */
+@DisableCachingByDefault(because = 'This is a disk bound copy/merge task.')
+@CompileStatic
+abstract class RoomSchemaLocationMergeTask extends DefaultTask {
+
+    // Using older internal API to maintain compatibility with Gradle 5.x
```
Comment (runningcode) [inline src/main/groovy/org/gradle/android/workarounds/room/task/RoomSchemaLocationMergeTask.groovy:25]: just a note, we don't need to worry about Gradle 5.x compatibility anymore

**Pull request merged**